client/gitSensor/GitSensorGrpcClient.go

```diff
@@ -0,0 +1,710 @@
+
+		selector := &WebhookEventSelectors{
+			Id:               int(item.Id),
+			EventId:          int(item.EventId),
+			Name:             item.Name,
+			ToShow:           item.ToShow,
+			ToShowInCiFilter: item.ToShowInCiFilter,
+			FixValue:         item.FixValue,
+			PossibleValues:   item.PossibleValues,
+			IsActive:         item.IsActive,
+		}
+		if item.CreatedOn != nil {
+			selector.CreatedOn = item.CreatedOn.AsTime()
+		}
+		if item.UpdatedOn != nil {
+			selector.UpdatedOn = item.UpdatedOn.AsTime()
+		}
+		selectors = append(selectors, selector)
+	}
+
+	mappedConfig := &WebhookEventConfig{
+		Id:            int(config.Id),
+		GitHostId:     int(config.GitHostId),
+		Name:          config.Name,
+		EventTypesCsv: config.EventTypesCsv,
+		ActionType:    config.ActionType,
+		IsActive:      config.IsActive,
+		Selectors:     selectors,
+	}
+	if config.CreatedOn != nil {
+		mappedConfig.CreatedOn = config.CreatedOn.AsTime()
+	}
+	if config.UpdatedOn != nil {
+		mappedConfig.UpdatedOn = config.UpdatedOn.AsTime()
+	}
+	return mappedConfig
+}
+
+// mapGitCommitToLocalType maps the protobuf specified GitCommit to local specified golang based struct
+func (client *GrpcApiClientImpl) mapGitCommitToLocalType(commit *pb.GitCommit) GitCommit {
+
+	mappedCommit := GitCommit{
+		Commit:  commit.Commit,
+		Author:  commit.Author,
+		Message: commit.Message,
+		Changes: commit.Changes,
+	}
+	if commit.Date != nil {
+		mappedCommit.Date = commit.Date.AsTime()
+	}
+	if commit.WebhookData != nil {
+		mappedCommit.WebhookData = &WebhookData{
+			Id:              int(commit.WebhookData.Id),
+			EventActionType: commit.WebhookData.EventActionType,
+			Data:            commit.WebhookData.Data,
+		}
+	}
+	return mappedCommit
+}
+
+func (client *GrpcApiClientImpl) mapWebhookEventConfigToProtoType(config *WebhookEventConfig) *pb.WebhookEventConfig {
+
+	selectors := make([]*pb.WebhookEventSelectors, 0)
+	for _, item := range config.Selectors {
+
+		selector := &pb.WebhookEventSelectors{
+			Id:               int64(item.Id),
+			EventId:          int64(item.EventId),
+			Name:             item.Name,
+			ToShow:           item.ToShow,
+			ToShowInCiFilter: item.ToShowInCiFilter,
+			FixValue:         item.FixValue,
+			PossibleValues:   item.PossibleValues,
+			IsActive:         item.IsActive,
+		}
+		if !item.CreatedOn.IsZero() {
+			selector.CreatedOn = timestamppb.New(item.CreatedOn)
+		}
+		if !item.UpdatedOn.IsZero() {
+			selector.UpdatedOn = timestamppb.New(item.UpdatedOn)
+		}
+		selectors = append(selectors, selector)
+	}
+
+	mappedConfig := &pb.WebhookEventConfig{
+		Id:            int64(config.Id),
+		GitHostId:     int64(config.GitHostId),
+		Name:          config.Name,
+		EventTypesCsv: config.EventTypesCsv,
+		ActionType:    config.ActionType,
+		IsActive:      config.IsActive,
+		Selectors:     selectors,
+	}
+	if !config.CreatedOn.IsZero() {
+		mappedConfig.CreatedOn = timestamppb.New(config.CreatedOn)
+	}
+	if !config.UpdatedOn.IsZero() {
```

Comment:
we can extract this logic and create a new wrapper method

Reply:
can be discussed